packages/design-system/src/components/contextMenu/menu.js

```diff
@@ -86,21 +87,25 @@ MenuWrapper.propTypes = {
 /**
  * Extracts all focusable children from an html tree, optionally ignoring items from submenu.
  *
- * @param {HTMLElement} parent The parent to search
- * @param {boolean} isSubMenu If we're searching from submenu.
+ * @param {Object} props Props.
+ * @param {HTMLElement} props.parent The parent to search
+ * @param {boolean} props.isSubMenu If we're searching from submenu.
+ * @param {boolean} props.wrapperRole the role assigned to the parent menu (sometimes a toolbar not a menu).
  * @return {Array.<HTMLElement>} List of focusable elements
  */
-function getFocusableChildren(parent, isSubMenu) {
+function getFocusableChildren({ parent, isSubMenu, wrapperRole = 'menu' }) {
   const allButtons = Array.from(
     parent.querySelectorAll(FOCUSABLE_SELECTORS.join(', '))
   );
   if (isSubMenu) {
     return allButtons;
   }
-  // Skip considering the submenu, and the submenu items.
-  return allButtons.filter(
-    (elem) => !elem.matches('[role="menu"], [role="menu"] [role="menu"] *')
-  );
+  // Skip considering the submenu, and the submenu items as well as inputs that are focus traps (floating menu/toolbar)
+  return allButtons.filter((elem) => {
+    return !elem.matches(
+      `[role="${wrapperRole}"], [role="${wrapperRole}"] [role="menu"] *, .${CONTEXT_MENU_SKIP_ELEMENT}`
+    );
+  });
```

Comment:
Adding this class `CONTEXT_MENU_SKIP_ELEMENT` to tell the contextMenu to skip over any elements with that class even if they are focusable we want to exclude them from our navigation. Fixes the various focus traps we've got goin on in the floating menu.